gobblin-core/src/main/java/org/apache/gobblin/converter/json/JsonStringToJsonIntermediateConverter.java

```diff
@@ -76,40 +79,483 @@ public JsonArray convertSchema(String inputSchema, WorkUnitState workUnit) throw
+      return element.getAsJsonPrimitive();
+    }
+    if (element.isJsonObject()) {
+      return element.getAsJsonObject();
+    }
+    if (element.isJsonNull()) {
+      return element.getAsJsonNull();
+    }
+    return null;
+  }
+
+  /**
+   * Wraps the object into a JsonObject with key temp
+   * @param prop
+   * @return
+   */
+  private JsonObject buildTempObject(JsonElement prop) {
+    JsonObject tempObject = new JsonObject();
+    if (prop instanceof JsonObject) {
+      tempObject.add(TEMP_COLUMN_NAME, prop.getAsJsonObject());
+    } else if (prop instanceof JsonArray) {
+      tempObject.add(TEMP_COLUMN_NAME, prop.getAsJsonArray());
+    } else if (prop instanceof JsonPrimitive) {
+      tempObject.add(TEMP_COLUMN_NAME, prop.getAsJsonPrimitive());
+    } else if (prop instanceof JsonNull) {
+      tempObject.add(TEMP_COLUMN_NAME, prop.getAsJsonNull());
+    }
+    return tempObject;
+  }
+
+  /**
+   * Wraps the schema within dataType and add columnName temp
+   * @param dataType
+   * @return
+   */
+  private JsonArray buildTempSchema(JsonObject dataType) {
+    JsonObject schemaObj = new JsonObject();
+    schemaObj.addProperty("columnName", TEMP_COLUMN_NAME);
+    schemaObj.add("dataType", dataType);
+    return jsonArray(schemaObj);
+  }
+
+  public static class JsonUtils {
+    /**
+     * Fetches dataType.values from the JsonObject
+     * @param schemaObject
+     * @return
+     */
+    public static JsonElement getValuesWithinDataType(JsonObject schemaObject) {
+      return getDataType(schemaObject).get("values");
+    }
+
+    /**
+     * Gets size for fixed type viz dataType.size from the JsonObject
+     * @param schemaObject
+     * @return
+     */
+    public static int getSizeOfFixedData(JsonObject schemaObject) {
+      return getDataType(schemaObject).get("size").getAsInt();
+    }
+
+    /**
+     * Gets schema of record which is present within array schema.
+     * @param schemaObject
+     * @return
+     */
+    public static JsonArray getSchemaForArrayHavingRecord(JsonObject schemaObject) {
+      return getValuesWithinDataType(getItems(getDataType(schemaObject)).getAsJsonObject()).getAsJsonArray();
+    }
+
+    /**
+     * Gets allowed symbols in enum from schema of enum type
+     * @param schemaObject
+     * @return
+     */
+    public static JsonArray allowedSymbolsInEnum(JsonObject schemaObject) {
+      return getDataType(schemaObject).get("symbols").getAsJsonArray();
+    }
+
+    public static boolean isEnumType(JsonObject schemaObject) {
+      return getDataTypeTypeFromSchema(schemaObject).get(0).getAsString().equalsIgnoreCase("enum");
+    }
+
+    /**
+     * Get dataType.type from schema
+     * @param schemaObject
+     * @return
+     */
+    public static JsonArray getDataTypeTypeFromSchema(JsonObject schemaObject) {
+      JsonObject dataType = getDataType(schemaObject);
+      return dataType.has("type") ? dataType.get("type").isJsonPrimitive() ? jsonArray(dataType.get("type"))
```

Comment:
Using nested operators like this is hard to read, I think an if statement would be better.